text/entities/0256-entities-data-model.md

```diff
@@ -0,0 +1,790 @@
+  global identification of an entity is available at once, in one place. This knowledge
+  may be distributed across multiple participants and needs to be combined to form an
+  identifier that is globally unique.
+
+- Semantic conventions must bring as much order as possible to telemetry, however they
+  cannot be too rigid and prevent real-world use cases.
+
+## Data Model
+
+We propose a new concept of Entity.
+
+Entity represents an object of interest associated with produced telemetry:
+traces, metrics or logs.
+
+For example, telemetry produced using OpenTelemetry SDK is normally associated with
+a Service entity. Similarly, OpenTelemetry defines system metrics for a host. The Host is the
+entity we want to associate metrics with in this case.
+
+Entities may be also associated with produced telemetry indirectly.
+For example a service that produces
+telemetry is also related with a process in which the service runs, so we say that
+the Service entity is related to the Process entity. The process normally also runs
+on a host, so we say that the Process entity is related to the Host entity.
+
+Note: subsequent OTEPs will define how the entities are associated with
+traces, metrics and logs and how relations between entities will be specified.
+See [Future Work](#future-work).
+
+The data model below defines a logical model for an entity (irrespective of the physical
+format and encoding of how entity data is recorded).
+
+<table>
+   <tr>
+    <td><strong>Field</strong>
+    </td>
+    <td><strong>Type</strong>
+    </td>
+    <td><strong>Description</strong>
+    </td>
+   </tr>
+   <tr>
+    <td>Type
+    </td>
+    <td>string
+    </td>
+    <td>Defines the type of the entity. MUST not change during the
+lifetime of the entity. For example: "service" or "host". This field is
+required and MUST not be empty for valid entities.
+    </td>
+   </tr>
+   <tr>
+    <td>Id
+    </td>
+    <td>map&lt;string, attribute&gt;
+    </td>
+    <td>Attributes that identify the entity.
+<p>
+MUST not change during the lifetime of the entity. The Id must contain
+at least one attribute.
+<p>
+Follows OpenTelemetry <a
+href="https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/common#attribute">common
+attribute definition</a>. SHOULD follow OpenTelemetry <a
+href="https://github.com/open-telemetry/semantic-conventions">semantic
+conventions</a> for attributes.
+    </td>
+   </tr>
+   <tr>
+    <td>Attributes
+    </td>
+    <td>map&lt;string, any&gt;
+    </td>
+    <td>Descriptive (non-identifying) attributes of the entity.
+<p>
+MAY change over the lifetime of the entity. MAY be empty. These
+attributes are not part of entity's identity.
+<p>
+Follows <a
+href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#type-any">any</a>
+value definition in the OpenTelemetry spec - it can be a scalar value,
```

Comment:
```suggestion
value definition in the OpenTelemetry log spec - it can be a scalar value,
```

Reply:
This means logs attributes will be mixed with the common attributes in the Id in the same type, right?

I see we capture this in the "open questions", but it isn't clear from that why we are doing this?  Is there a reason we need the log attribute types here?

Reply:
We want to limit Id key/value pairs to common attribute types. The reason for this is that Id key/values will be also duplicated as Resource attributes (more on this in the next OTEP). We control what the Id is and we can place this restriction.

As opposed to that we often don't control Entity's descriptive attributes. The information about the Entity may originate externally and have arbitrary complex structure. We need to be able to record this information in its original, structured form. This requirement is exactly similar to what we have for logs, where log attributes may also originate externally.

I am not particularly happy about this situation, that's why the open question. We will work on this and will try to find a better approach.